operator/controllers/lifecycle/keptntask/container_builder.go

```diff
@@ -1,46 +1,35 @@
 package keptntask
 
 import (
-	"reflect"
-
-	"github.com/go-logr/logr"
 	klcv1alpha3 "github.com/keptn/lifecycle-toolkit/operator/apis/lifecycle/v1alpha3"
 	"golang.org/x/net/context"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/client-go/tools/record"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-// IContainerBuilder is the interface that describes the operations needed to help build job specs of a task
-type IContainerBuilder interface {
-	// CreateContainerWithVolumes returns a job container and volumes based on the task definition spec
-	CreateContainerWithVolumes(ctx context.Context) (*corev1.Container, []corev1.Volume, error)
-}
-
-// BuilderOptions contains everything needed to build the current job
-type BuilderOptions struct {
-	client.Client
-	recorder record.EventRecorder
-	req      ctrl.Request
-	Log      logr.Logger
-	task     *klcv1alpha3.KeptnTask
-	taskDef  *klcv1alpha3.KeptnTaskDefinition
+// JSBuilder implements container builder interface for javascript deno
+type ContainerBuilder struct {
+	taskDef *klcv1alpha3.KeptnTaskDefinition
 }
 
-func getContainerBuilder(options BuilderOptions) IContainerBuilder {
-	if isJSSpecDefined(&options.taskDef.Spec) {
-		builder := newJSBuilder(options)
-		return &builder
+func newContainerBuilder(taskDef *klcv1alpha3.KeptnTaskDefinition) *ContainerBuilder {
+	return &ContainerBuilder{
+		taskDef: taskDef,
 	}
-	return nil
 }
 
-func specExists(definition *klcv1alpha3.KeptnTaskDefinition) bool {
-	//TODO when adding new builders add more logic here
-	return isJSSpecDefined(&definition.Spec)
-}
-
-func isJSSpecDefined(spec *klcv1alpha3.KeptnTaskDefinitionSpec) bool {
-	return !reflect.DeepEqual(spec.Function, klcv1alpha3.FunctionSpec{})
+func (c *ContainerBuilder) CreateContainerWithVolumes(ctx context.Context) (*corev1.Container, []corev1.Volume, error) {
+	jobVolumes := []corev1.Volume{
+		{
+			Name: "default-volume",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					// Default 50% of the memory of the node, max 1Gi
+					SizeLimit: resource.NewQuantity(1, resource.Format("Gi")),
+					Medium:    corev1.StorageMedium("Memory"),
+				},
+			},
+		},
+	}
+	return c.taskDef.Spec.Container.Container, jobVolumes, nil
```

Comment:
I find this weird. why do we return a field of the task here? it is untouched

Reply:
I see later that in the function builder concerns are not split. I think we can refactor everything into a more cleaner separation of concerns making the JSBuilder logic easier to follow. This can be done in a follow-up

Reply:
agree, will create a follow-up ticket

Reply:
created a follow-up ticket https://github.com/keptn/lifecycle-toolkit/issues/1500